File: src/main/java/com/dashboard/services/UserService.java
package com.dashboard.services;

import com.dashboard.exceptions.UserNotFoundException;
import com.dashboard.models.JPA.User;
import com.dashboard.repositories.UserRepository;
import com.dashboard.interfaces.serviceInterfaces.UserServiceInterface;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class UserService implements UserServiceInterface {

    @Autowired
    private UserRepository userRepository;

    @Override
    public void save(User user) {
        userRepository.save(user);
    }

    @Override
    public void delete(Integer userId) {
        final Optional<User> byId = userRepository.findById(userId);
        if(byId.isEmpty()) throw new UserNotFoundException();
        userRepository.deleteById(userId);
    }

    @Override
    public void update(Integer userId, User user) {
        final Optional<User> byId = userRepository.findById(userId);
        if(byId.isEmpty()) throw new UserNotFoundException();
        user.setId(userId);
        userRepository.deleteById(userId);
        userRepository.save(user);
    }

    @Override
    public boolean exist(Integer userId) {
        return userRepository.findById(userId).isPresent();
    }

    @Override
    public List<User> getAllUsers() {
        return userRepository.findAll();
    }

    @Override
    public User getUserById(Integer userId) {
        final Optional<User> byId = userRepository.findById(userId);
        if(byId.isEmpty()) throw new UserNotFoundException();
        return byId.get();
    }
}

File: src/main/kotlin/dashboard/services/UserService.kt
package dashboard.services

import dashboard.exceptions.UserNotFoundException
import dashboard.interfaces.serviceInterfaces.UserServiceInterface
import dashboard.models.jpa.User
import org.springframework.stereotype.Service
import dashboard.repositories.UserRepository
import java.util.*

@Service
class UserService(val userRepository: UserRepository) : UserServiceInterface {
    
    override fun save(user: User) {
        userRepository.save(user)
    }

    override fun delete(userId: Int) {
        val byId: Optional<User> = userRepository.findById(userId)
        if (byId.isEmpty) throw UserNotFoundException()
        userRepository.deleteById(userId)
    }

    @Throws(UserNotFoundException::class)
    override fun update(userId: Int, user: User) {
        val byId: Optional<User> = userRepository.findById(userId)
        if (byId.isEmpty) throw UserNotFoundException()
        user.id = userId
        userRepository.deleteById(userId)
        userRepository.save(user)
    }

    override fun exist(userId: Int): Boolean {
        return userRepository.findById(userId).isPresent
    }

    override val allUsers: List<User>
        get() = userRepository.findAll()

    override fun getUserById(userId: Int): User {
        val byId: Optional<User> = userRepository.findById(userId)
        if (byId.isEmpty) throw UserNotFoundException()
        return byId.get()
    }
}